api/core/v1beta2/cluster_types.go

```diff
@@ -1137,6 +1137,7 @@ func (c *ClusterStatus) GetTypedPhase() ClusterPhase {
 // ANCHOR: APIEndpoint
 
 // APIEndpoint represents a reachable Kubernetes API endpoint.
+// +kubebuilder:validation:MinProperties=1
```

Comment:
Has two required fields? Shouldn't need this.

The fields that are required must set a minlength/minimum, as `""`/`0` is not really valid right?

Reply:
I think this change should not be part of this PR, we'll address that separately (that's why I was surprised that this PR adds MinProperties :))